scenes/game/local_game.gd

```diff
@@ -4314,7 +4352,9 @@ func handle_strike_effect(card_id :int, effect, performing_player : Player):
 			_append_log_full(Enums.LogType.LogType_Effect, performing_player, "'s %s is no longer blocking opponent movement." % buddy_name)
 		"return_attack_to_top_of_deck":
 			performing_player.strike_stat_boosts.attack_to_topdeck_on_cleanup = true
-			handle_strike_attack_immediate_removal(performing_player)
+			events += handle_strike_attack_immediate_removal(performing_player)
```

Comment:
is this really how hadoken works (i.e. losing all stats if you loop)? i guess it doesn't generally come up since it doesn't have armor/guard

Reply:
yes